messaging/src/main/java/io/atomix/messaging/impl/NettyMessagingService.java

```diff
@@ -525,14 +525,16 @@ private Bootstrap bootstrapClient(Endpoint endpoint) {
     Bootstrap bootstrap = new Bootstrap();
     bootstrap.option(ChannelOption.ALLOCATOR, PooledByteBufAllocator.DEFAULT);
     bootstrap.option(ChannelOption.WRITE_BUFFER_WATER_MARK,
-        new WriteBufferWaterMark(10 * 32 * 1024, 10 * 64 * 1024));
-    bootstrap.option(ChannelOption.SO_SNDBUF, 1048576);
+            new WriteBufferWaterMark(10 * 32 * 1024, 10 * 64 * 1024));
+    bootstrap.option(ChannelOption.SO_RCVBUF, 1024 * 1024);
+    bootstrap.option(ChannelOption.SO_SNDBUF, 1024 * 1024);
+    bootstrap.option(ChannelOption.SO_KEEPALIVE, true);
+    bootstrap.option(ChannelOption.TCP_NODELAY, true);
```

Comment:
Nagles must be disabled on client and server for it to work.

Reply:
Nice